services/requester/requester.go

```diff
@@ -597,7 +526,111 @@ func (e *EVM) dryRunTx(
 	return result, nil
 }
 
-func AddOne64th(n uint64) uint64 {
-	// NOTE: Go's integer division floors, but that is desirable here
-	return n + (n / 64)
+// buildTransaction creates a flow transaction from the provided script,
+// with the arguments and signs it with the configured COA account.
+func (e *EVM) buildTransaction(
+	ctx context.Context,
+	script []byte,
+	args ...cadence.Value,
+) (*flow.Transaction, error) {
+	// building and signing transactions should be blocking, so we don't have keys conflict
+	e.mux.Lock()
```

Comment:
**Note:** I think we could get rid of the lock acquisition here, as the key management is now inside the `Keystore`, and that is already blocking.